docs/react-testing-library/migrate-from-enzyme.md

```diff
@@ -0,0 +1,193 @@
+// Your tests come here...
+   });
+})
+
+```
+
+### Basic Enzyme to React Testing Library migration
+One thing to keep in mind that there's not a one-to-one mapping of Enzyme features to React Testing Library features. 
+Many Enzyme features result in inefficient tests, and we are here to learn the React Testing Library’s philosophy and 
+testing approach.
+
+React Testing Library has dozens of helpful queries which lets you access your component’s elements and their 
+properties, and here we’ll show you a very basic Enzyme’s queries with React Testing Library’s alternatives.
+
+Let’s say we have a `Welcome` component, which just shows a welcome message, and we will have a look at both Enzyme 
+and React Testing Library tests to learn how we can test this component:
+
+**Enzyme**
+```
+it('should have correct welcome text', ()=>{
+   const wrapper = shallow(<Welcome  name='John Doe'/>);
+   expect(wrapper.text()).to.equal('Welcome, John Doe');   
+})
+```
+
+
+**React Testing library**
+```
+ it("should have correct welcome text", () => {
+   render(<Welcome name='John Doe'/>);
+   expect(screen.getByText('Welcome, John Doe')).toBeInTheDocument();
+});
+```
+Alright! as you see, both are pretty similar, Enzyme's `shallow` wrapping doesn’t descend down to sub-components, 
+React Testing Library’s `render` is more similar to `mount`.
+
+You don’t need to assign the `render` result to a variable (ex wrapper, or etc), and you can simply access the 
+rendered output by calling the `screen`. The other good thing to know is that React Testing Library automatically 
+cleans up the output for each test, so you don’t need to call `cleanup` on Jest’s `AfterEeach` function.
+
+React Testing Library aims to test the component, like how users would, users see the button, heading, and other
+elements by their role, not by their `id` or `class`, or element tag name. When you use React Testing Library, you 
+should not try to access the DOM like how you would do by JQuery API. We made some handy which help you to access 
+your elements very efficiently. 
+
+You can see the [list of available queries](https://testing-library.com/docs/dom-testing-library/api-queries) in detail. 
+Something else that people have a problem with is that they’re not sure which query should they use, luckily we have 
+a great page which explains [which query to use](https://testing-library.com/docs/guide-which-query), so don’t forget 
+to check it out!
+
+If you still have a problem with the React Testing Library’s queries, and you’re not sure which query to use, just 
+don’t worry! we’ve created a useful Chrome extension named 
+**[which-query](https://chrome.google.com/webstore/detail/testing-library-which-que/olmmagdolfehlpjmbkmondggbebeimoh)**  
+that aims to enable developers to find a better query when writing tests. Right-click on an element on any web page 
+and it will copy the best possible query for that element into your clipboard. Then simply paste into your test code. 
```

Comment:
I think we're recommending Testing Playground instead.

Reply:
You're right. I'll change that too